dcos/dcos-ui
# feat(DCOS-14986): i18n poc implementation.

This PR is a poc implementation of i18n where two libs were explored (React-intl and i18next).

React-intl was chosen for the following reasons, support all acceptance criteria defined in `DCOS-14986`,
has been proven internally and externally at Yahoo, community support, follows React-like pattern where components are used instead imperative way (calling functions),
Good documentation/API/source code and lastly follows the ECMA standards for i18n. Unfortunately both contain only one main contributor which ins't ideal but since Yahoo has a better open source history (e.g YUI) make it "safer" to use.

We also have other team members (@jfurrow) familiar with React-intl which makes it faster to adopt.

This PR also include the documentation guide to get started on how to add new translations (files or key/values) and a couple of other points where to find more information about the React-Intl.

Steps to test:
`npm i react-intl` then `npm start` when the build is ready you can check the `dashboard` page `panels heading` and `view all` buttons

**before**
<img width="1185" alt="before" src="https://cloud.githubusercontent.com/assets/549394/24731501/5ff3b232-1a1f-11e7-977d-9dc47409f4f3.png">

**after**
<img width="1169" alt="after" src="https://cloud.githubusercontent.com/assets/549394/24731507/67f499c4-1a1f-11e7-9c03-073f9810145e.png">


**Checklist**
